universal-login-relayer/lib/services/MessageHandler.ts

```diff
@@ -1,34 +1,48 @@
 import {Wallet, providers} from 'ethers';
 import {EventEmitter} from 'fbemitter';
 import {SignedMessage} from '@universal-login/commons';
-import {isAddKeyCall, getKeyFromData, isAddKeysCall, getRequiredSignatures, messageToTransaction} from '../utils/utils';
+import {isAddKeyCall, getKeyFromData, isAddKeysCall, getRequiredSignatures} from '../utils/utils';
 import AuthorisationService from './authorisationService';
 import TransactionQueueService from './transactions/TransactionQueueService';
 import PendingMessages from './messages/PendingMessages';
-import {ensureEnoughToken, ensureEnoughGas} from './messages/validations';
 import {decodeDataForExecuteSigned} from './transactions/serialisation';
+import TransactionExecutor from './transactions/TransactionExecutor';
 
 class TransactionService {
+  private executor: TransactionExecutor;
 
-  constructor(private wallet: Wallet, private authorisationService: AuthorisationService, private hooks: EventEmitter, private provider: providers.Provider, private transactionQueue: TransactionQueueService, private pendingMessages: PendingMessages) {
+  constructor(private wallet: Wallet, private authorisationService: AuthorisationService, private hooks: EventEmitter, private transactionQueue: TransactionQueueService, private pendingMessages: PendingMessages) {
+    this.executor = new TransactionExecutor(
+        this.wallet,
+        this.onTransactionSent(
+         this.hooks,
+         this.removeReqFromAuthService(this.authorisationService)
+        )
+      );
   }
 
   start() {
-    this.transactionQueue.setOnTransactionSent(this.onTransactionSent);
+    this.transactionQueue.setOnTransactionSent(
```

Comment:
Why not just:
`this.transactionQueue.setOnTransactionSent(this.onTransactionSent);`